### PR TITLE
[WIP]add e2e for tidbinitializer controller

### LINKED ISSUE #1404: UT & e2e test for TidbInitializer controller
## Feature Request

**Is your feature request related to a problem? Please describe:**

This issue is to trace the UT & e2e test for PR https://github.com/pingcap/tidb-operator/pull/1403

### PR DESCRIPTION
**Pull request by DanielZhangQD**
### What problem does this PR solve? 
fix #1404
### What is changed and how does it work?
add e2e & UT for tidbinitializer controller
### Check List 

Tests 

 - Unit test
 - E2E test

Code changes
 - Has Go code change

### Does this PR introduce a user-facing change?:
 
 ```release-note
NONE
 ```

### CODE REVIEW

File: tests/actions.go [hunk 1]
```diff
@@ -678,6 +688,93 @@ func (oa *operatorActions) DeployTidbClusterOrDie(info *TidbClusterConfig) {
...
+		return fmt.Errorf("failed to create secret of cluster [%s]: %v", info.ClusterName, err)
+	}
+
+	err = oa.CreateConfigMap(info)
+	if err != nil {
+		return fmt.Errorf("failed to create configmap of cluster [%s]: %v", info.ClusterName, err)
+	}
+
+	cmd := fmt.Sprintf("helm install %s  --name %s --namespace %s --set-string %s",
+		oa.tidbClusterChartPath(info.OperatorTag), info.ClusterName, info.Namespace, info.TidbClusterHelmSetString(nil, false))
```
Comment (cofyc) [inline tests/actions.go]: don't use helm anymore

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 2]
```diff
@@ -686,6 +686,39 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
 		err = oa.WaitForTidbClusterReady(tc, 30*time.Minute, 15*time.Second)
 		framework.ExpectNoError(err, "Expected TiDB cluster scaled in and ready")
 	})
+	ginkgo.It("TiDBInitializer: Test managing tidb initializer job with TidbInitializer CRD", func() {
+		cluster := newTidbClusterConfig(e2econfig.TestConfig, ns, "init", "admin", "")
+		cluster.Resources["pd.replicas"] = "1"
+		cluster.Resources["tikv.replicas"] = "1"
+		cluster.Resources["tidb.replicas"] = "1"
+		oa.DeployTidbClusterForInitOrDie(&cluster)
```
Comment (cofyc) [inline tests/e2e/tidbcluster/tidbcluster.go]: I don't think it's necessary to add any complex methods to `operatorActions`. Right now, it's ok to deploy tidb cluster with plain Kubernetes client.  Here is an example to deploy a tidb cluster and wait for it to be ready. https://github.com/pingcap/tidb-operator/blob/master/tests/e2e/tidbcluster/tidbcluster.go#L654-L688

If you want to update tidb cluster CRD object, you can modify its fixture function or update before creating.

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 3]
```diff
@@ -686,6 +686,39 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
...
+
+		ti := fixture.GetTidbInitializer(ns, "init", "init", "init-set-srcret", cluster.Resources["tidb.initSql"], "", corev1.PullIfNotPresent)
+		oa.DeployTidbInitializerOrDie(&cluster, ti)
```
Comment (cofyc) [inline tests/e2e/tidbcluster/tidbcluster.go]: I think one create request should be enough.

File: tests/actions.go [hunk 4]
```diff
@@ -678,6 +688,93 @@ func (oa *operatorActions) DeployTidbClusterOrDie(info *TidbClusterConfig) {
...
+
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ns,
+		},
+	}
+	_, err := oa.kubeCli.CoreV1().Namespaces().Create(namespace)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("failed to create namespace[%s]:%v", ns, err)
+	}
```
Comment (cofyc) [inline tests/actions.go]: no need to create the namespace, in ginkgo, each test has a namespace

**Comment by Yisaer** [comment]
@DanielZhangQD  Is this still WIP?

**Comment by DanielZhangQD** [comment]
> @DanielZhangQD Is this still WIP?

Yes, sorry for the delay, please help finish the test case, thanks a lot!

**Pull request closed without merging**